libs/langchain/langchain/llms/vertexai.py

```diff
@@ -393,6 +395,10 @@ def validate_environment(cls, values: Dict) -> Dict:
             raise ValueError(
                 "A GCP project should be provided to run inference on Model Garden!"
             )
+        if values["location"] is None:                 
+            raise ValueError(
```

Comment:
this is a breaking change, don't think it's worth making

Reply:
hardcoding a location is not a good practice to begin with. I replaced location: str to location: Optional[str] to make sure that this is not a breaking change. location is only being used in VertexAIModelGarden class

Reply:
Nothing is hard coded here, it's just a default value. so +1 to @baskaryan

Optional hint means that a value can be None, and I don't think it's the case here.

https://peps.python.org/pep-0484/#union-types

Reply:
the original form is location: "us-central1" so this is hardcoding. As the code stands location is being utilised only in VertexAIModelGarden class and it is a required parameter. Because of this I also added location to the     @root_validator decorator to make sure that the user supplies this information. and I know what Optional hint means.

Reply:
Even if it is a breaking change, this is a breaking change for something that is not working today anyways. the existing code doesnt work anyways